### PR TITLE
repurpose "name" property, add displayName* props instead

### PR DESCRIPTION
**Pull request by mzanetti**

### CODE REVIEW

**Comment by mzanetti** [comment]
silo this!

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
Jenkins, test this please

**Comment by mzanetti** [comment]
Jenkins, retest this please

File: libguh/plugin/deviceplugin.cpp [hunk 1]
```diff
@@ -596,16 +605,30 @@ void DevicePlugin::loadMetaData()
...
                 QVariant::Type t = QVariant::nameToType(st.value("type").toString().toLatin1().data());
+                if (t == QVariant::Invalid) {
+                    qCWarning(dcDeviceManager()) << "Invalid StateType type:" << st.value("type").toString();
+                    broken = true;
+                    break;
+                }
                 StateType stateType(st.value("id").toString());
-                stateType.setName(translateValue(m_metaData.value("idName").toString(), st.value("name").toString()));
+                stateType.setName(st.value("name").toString());
+                qWarning() << "******** translation:" << m_metaData.value("name").toString() << st.value("displayName").toString() << translateValue(m_metaData.value("name").toString(), st.value("displayName").toString());
```
Comment (t-mon) [inline libguh/plugin/deviceplugin.cpp]: Use a debug category and keep the debug message consistent (Remove `********` and start messages with a capital letter)

File: libguh/types/statetype.cpp [hunk 2]
```diff
@@ -49,18 +49,30 @@ StateTypeId StateType::id() const
     return m_id;
 }
 
-/*! Returns the name of the StateType. This is visible to the user (e.g. "Temperature"). */
+/*! Returns the name of the StateType. This is used internally, e.g. to match interfaces. */
```
Comment (t-mon) [inline libguh/types/statetype.cpp]: We could create a link to the Interfaces page here

File: plugins/mock/devicepluginmock.cpp [hunk 3]
```diff
@@ -320,7 +369,7 @@ void DevicePluginMock::emitDevicesDiscovered()
     if (m_discoveredDeviceCount > 1) {
         DeviceDescriptor d2(mockDeviceClassId, "Mock Device 2 (Discovered)", "55556");
         ParamList params;
-        Param httpParam(httpportParamTypeId, "55556");
+        Param httpParam(mockHttpportParamTypeId, "55556");
```
Comment (t-mon) [inline plugins/mock/devicepluginmock.cpp]: Optional: `mockHttpPortParamTypeId`

**Comment by mzanetti** [comment]
Jenkins, retest this please

**Comment by mzanetti** [comment]
Jenkins, land this pleaser

**Pull request merged**